docs/usage/middleware/builtin-middleware.rst

```diff
@@ -42,29 +42,74 @@ This middleware prevents CSRF attacks by doing the following:
 
 1. On the first "safe" request (e.g GET) - set a cookie with a special token created by the server
 2. On each subsequent "unsafe" request (e.g POST) - make sure the request contains either a
-    form field or an additional header that has this token
-
+    form field or an additional header that has this token (more on this below)
 
 To enable CSRF protection in a Litestar application simply pass an instance of
 :class:`CSRFConfig <.config.csrf.CSRFConfig>` to the Litestar constructor:
 
 .. code-block:: python
 
-    from litestar import Litestar
+    from litestar import Litestar, get, post
     from litestar.config.csrf import CSRFConfig
 
+
+    @get()
+    async def get_resource() -> str:
+        # GET is one of the safe methods
+        return "some_resource"
+
+    @post("{id:int}")
+    async def create_resource(id: int) -> bool:
+        # POST is one of the unsafe methods
+        return True
+
     csrf_config = CSRFConfig(secret="my-secret")
 
-    app = Litestar(route_handlers=[...], csrf_config=csrf_config)
+    app = Litestar([get_resource, create_resource], csrf_config=csrf_config)
 
 
-Routes can be marked as being exempt from the protection offered by this middleware via
-:ref:`handler opts <handler_opts>`
+The following snippet demonstrates how to change the cookie name to "some-cookie-name" and header name to "some-header-name".
 
 .. code-block:: python
 
-    from litestar import post
+    csrf_config = CSRFConfig(secret="my-secret", cookie_name='some-cookie-name', header_name='some-header-name')
+
+
+A CSRF protected route can be accessed by any client that can make a request with either the header or form-data key.
+
+
+.. note::
+
+    The form-data key can not be currently configured. It should only be passed via the key "_csrf_token"
+
+In Python, any client such as `requests <https://github.com/psf/requests>`_ or `httpx <https://github.com/encode/httpx>`_ can be used.
+The usage of clients or sessions is recommended due to the cookie persistence it offers across requests.
+The following is an example using ``httpx.Client``.
+
+.. code-block:: python
 
+    import httpx
+
+
+    with httpx.Client() as client:
+        get_response = client.get("http://localhost:8000/")
+
+        # "csrftoken" is the default cookie name
+        csrf = get_response.cookies["csrftoken"]
```

Comment:
would it makes sense to add a post with a wrong header / cookie just to show the protection ?

Reply:
would this be ok?
```py
    # the following is a request that will fail because it does not have a cookie in its session
    post_response_with_no_persisted_cookie = httpx.post("http://localhost:8000/1", headers={"x-csrftoken": csrf})
    assert post_response_with_no_persisted_cookie.status_code == 403
    assert "CSRF token verification failed" in post_response_with_no_persisted_cookie.text
    ```

Reply:
yep, it shows well the protection, now I'm not sure this is not too detailed but imho it makes sense to show it working.

Reply:
I understand but I am unsure how else to approach this tbh, I feel if I make it more detailed it would just parrot the tests we already have, one can argue normal users won't look at tests, so I am open for improvements here 🙂

Reply:
> it makes sense to show it working

Would it? As a user I would assume that, by default, sending the wrong cookie/header wouldn't work.